Linking error Arch cannot find -laudio

On Arch WSL, building with `-DENABLE_SDL2_MIXER=OFF` compiles successfully into object code, but has a linking error. It cannot find libaudio.

```
/usr/sbin/ld: cannot find -laudio
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/Cytopia.dir/build.make:569: Cytopia] Error 1
make[1]: *** [CMakeFiles/Makefile2:361: CMakeFiles/Cytopia.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

Ok so I tested this on arch (not WSL), and I get the following error:

```
ERROR: Not able to automatically detect '/usr/sbin/c++' version
ERROR: Unable to find a working compiler
ERROR: sdl2/2.0.9@bincrafters/stable: Error in package_info() method, line 333
	self.add_libraries_from_pc('alsa')
while calling 'add_libraries_from_pc', line 304
	libs = [lib[2:] for lib in pkg_config.libs_only_l]  # cut -l prefix
	ConanException: pkg-config command ['pkg-config', '--libs-only-l', 'alsa', '--static', '--print-errors'] failed with error: Command 'pkg-config --libs-only-l alsa --static --print-errors' returned non-zero exit status 127.
CMake Error at install/_pmm/1.4.2/conan.cmake:475 (message):
  Conan install failed [1]:

Call Stack (most recent call first):
  install/_pmm/1.4.2/conan.cmake:500 (_pmm_conan_install_1)
  install/_pmm/1.4.2/conan.cmake:648 (_pmm_conan_install)
  install/_pmm/1.4.2/main.cmake:32 (_pmm_conan)
  install/_pmm/1.4.2/main.cmake:47 (_pmm_project_fn)
  CMakeLists.txt:56 (pmm)


CMake Error at install/_pmm/1.4.2/conan.cmake:505 (message):
  Conan dependencies were not imported (Expected file
  /Cytopia/install/conanbuildinfo.cmake).  You may need to run Conan manually
  (from the build directory).  Ensure you are using the 'cmake' generator.
Call Stack (most recent call first):
  install/_pmm/1.4.2/conan.cmake:648 (_pmm_conan_install)
  install/_pmm/1.4.2/main.cmake:32 (_pmm_conan)
  install/_pmm/1.4.2/main.cmake:47 (_pmm_project_fn)
  CMakeLists.txt:56 (pmm)
```

__How to reproduce__
```
docker run -it --rm archlinux/base bash -c "\
pacman -Syu --noconfirm &&\
pacman -S --noconfirm git cmake make gcc python3 &&\
git clone https://github.com/JimmySnails/Cytopia.git &&\
cd Cytopia &&\
mkdir install &&\
cd install &&\
cmake .. -DENABLE_SDL2_MIXER=Off &&\
make"
```

I think libalsa is a SDL2 dependency and I'm not sure why. Did we ever test if `ENABLE_SDL2_MIXER` works as expected?

--- a/conanfile.txt.in
+++ b/conanfile.txt.in
@@ -6,7 +6,8 @@ cmake
 sdl2/2.0.9@bincrafters/stable
 sdl2_ttf/2.0.15@bincrafters/stable
 sdl2_image/2.0.4@bincrafters/stable
-@_CC_SDL2_MIXER@sdl2_mixer/2.0.4@anotherfoxguy/stable
+bzip2/1.0.8@conan/stable
+@_CC_SDL2_MIXER@sdl2_mixer/2.0.4@bincrafters/stable 
 @_CC_MOFILEREADER@MofileReader/1.0.0@anotherfoxguy/stable
 libpng/1.6.36@bincrafters/stable
 freetype/2.9.1@bincrafters/stable
@@ -16,3 +17,4 @@ OpenSSL/1.1.1a@conan/stable
 
 [options]
 sdl2_mixer:opus=False # PACKAGE:OPTION=VALUE
+sdl2_mixer:fluidsynth=False

